GetStream/stream-chat-android
# [3369] Video thumbnails UI components (MessageList)

### 🎯 Goal

This PR implements the Message List part of the video thumbnails effort, the gallery part will arrive in a separate PR.

### 🛠 Implementation details

Reworked the image attachment ViewHolder and its child elements in order to display both image and video attachments.

### 🎨 UI Changes

| Before | After |
| --- | --- |
| ![Screenshot_1665158372](https://user-images.githubusercontent.com/37080097/194598267-8ef9d4e2-f41c-4e33-96dd-21885710c615.png) | ![Screenshot_1665158218](https://user-images.githubusercontent.com/37080097/194598274-06d57be8-b6b3-4cb2-873b-ff7d28793fb4.png)  |

### 🧪 Testing

Test A:

1. Launch the XML app
2. Go to a channel
3. test sending mixed attachments in patterns such as: single image, single video, single other attachment, multiple images, multiple videos, multiple other attachments, mixed image and video attachments and mixed various attachments.

Expectation: Everything should be rendered correctly with images and video being rendered as previewable attachments while other types will be rendered as file attachments.

Test B:

1. Apply given patch
4. Launch the XML app
5. Observe that the style of the media attachments has now changes, pay special attention to the play button
6. Feel free to customize this further yourself.

<details>

<summary>Provide the patch summary here</summary>

```
Index: stream-chat-android-ui-components/src/main/res/values/styles.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-ui-components/src/main/res/values/styles.xml b/stream-chat-android-ui-components/src/main/res/values/styles.xml
--- a/stream-chat-android-ui-components/src/main/res/values/styles.xml	(revision de1d49bceccd2bc6d21e13dd245e42f391a286a1)
+++ b/stream-chat-android-ui-components/src/main/res/values/styles.xml	(date 1665159510998)
@@ -507,16 +507,18 @@
     <style name="StreamUi.MessageList.MediaAttachment" parent="StreamUi">
         <item name="streamUiMediaAttachmentProgressIcon">@drawable/stream_ui_rotating_indeterminate_progress_gradient</item>
         <item name="streamUiMediaAttachmentPlaceHolderIcon">@drawable/stream_ui_picture_placeholder</item>
-        <item name="streamUiMediaAttachmentMediaPreviewBackgroundColor">@color/stream_ui_message_list_image_attachment_background</item>
+        <item name="streamUiMediaAttachmentPlaceHolderIconTint">@color/stream_ui_literal_white</item>
+        <item name="streamUiMediaAttachmentMediaPreviewBackgroundColor">@color/stream_ui_black_50</item>
         <item name="streamUiMediaAttachmentMoreCountOverlayColor">@color/stream_ui_overlay</item>
         <item name="streamUiMediaAttachmentPlayVideoIcon">@drawable/stream_ui_ic_play</item>
-        <item name="streamUiMediaAttachmentPlayVideoIconElevation">3dp</item>
-        <item name="streamUiMediaAttachmentPlayVideoIconPaddingTop">2dp</item>
-        <item name="streamUiMediaAttachmentPlayVideoIconPaddingBottom">2dp</item>
-        <item name="streamUiMediaAttachmentPlayVideoIconPaddingStart">3dp</item>
-        <item name="streamUiMediaAttachmentPlayVideoIconPaddingEnd">2dp</item>
-        <item name="streamUiMediaAttachmentPlayVideoIconCornerRadius">24dp</item>
-        <item name="streamUiMediaAttachmentPlayVideoIconBackgroundColor">@color/stream_ui_literal_white</item>
+        <item name="streamUiMediaAttachmentPlayVideoIconTint">@color/stream_ui_white</item>
+        <item name="streamUiMediaAttachmentPlayVideoIconElevation">0dp</item>
+        <item name="streamUiMediaAttachmentPlayVideoIconPaddingTop">4dp</item>
+        <item name="streamUiMediaAttachmentPlayVideoIconPaddingBottom">4dp</item>
+        <item name="streamUiMediaAttachmentPlayVideoIconPaddingStart">5dp</item>
+        <item name="streamUiMediaAttachmentPlayVideoIconPaddingEnd">4dp</item>
+        <item name="streamUiMediaAttachmentPlayVideoIconCornerRadius">5dp</item>
+        <item name="streamUiMediaAttachmentPlayVideoIconBackgroundColor">@color/stream_ui_black_50</item>
 
         <item name="streamUiMediaAttachmentMoreCountTextSize">@dimen/stream_ui_message_image_attachment_more_count_text_size</item>
         <item name="streamUiMediaAttachmentMoreCountTextColor">@color/stream_ui_literal_white</item>
```

</details>


### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist

### 🎉 GIF

![gif](https://media0.giphy.com/media/l46CdxUEWXLaBzMVW/giphy.gif?cid=ecf05e47k25qk2u2sil4l0b18ieipvj5f3tvv88zobbp2kht&rid=giphy.gif&ct=g)
